### PR TITLE
Update the new release steps

### PR DESCRIPTION
**Pull request by mimowo**
#### What type of PR is this?

/kind documentation
/kind cleanup

#### What this PR does / why we need it:

To maintain the release steps and make them as clear as possible. 

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

The update is based on experiences with https://github.com/kubernetes-sigs/kueue/issues/2985

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by mimowo** [comment]
/cc @alculquicondor @tenzen-y 
/hold
until 0.8.1 is out in case more updates needs to be done. Still, feel free to provide early feedback.

**Comment by alculquicondor** [comment]
LGTM so far

File: .github/ISSUE_TEMPLATE/NEW_RELEASE.md [hunk 1]
```diff
@@ -35,12 +35,13 @@ Please do not remove items from the checklist
       and inserts the changelog into the tag description.
```
Comment (alculquicondor) [inline .github/ISSUE_TEMPLATE/NEW_RELEASE.md:35]: When building/developing Kueue on a remote machine, it is easier to do things in a different order:
1. push the tag
2. create draft referencing the tag
3. use `gh release upload <tag> artifacts/*`

File: .github/ISSUE_TEMPLATE/NEW_RELEASE.md [hunk 2]
```diff
@@ -50,12 +51,17 @@ Please do not remove items from the checklist
 - [ ] For major or minor releases, merge the `main` branch into the `website` branch to publish the updated documentation.
 - [ ] Send an announcement email to `sig-scheduling@kubernetes.io` and `wg-batch@kubernetes.io` with the subject `[ANNOUNCE] kueue $VERSION is released`.   <!--Link: example https://groups.google.com/a/kubernetes.io/g/wg-batch/c/-gZOrSnwDV4 -->
 - [ ] Update the below files with respective values in `main` branch :
-  - Latest version in `README.md`
-  - Latest version in `cmd/experimental/kjobctl/docs/installation.md`
+  - Latest version in:
+    - `README.md`
+    - `Makefile`
```
Comment (alculquicondor) [inline .github/ISSUE_TEMPLATE/NEW_RELEASE.md:56]: just the makefile, and then run `make prepare-release-branch`

File: .github/ISSUE_TEMPLATE/NEW_RELEASE.md [hunk 3]
```diff
@@ -50,12 +51,17 @@ Please do not remove items from the checklist
...
+    - `installation.md`
+    - `cmd/experimental/kjobctl/docs/installation.md`
   - Release notes in the `CHANGELOG`
   - `version` in `site/hugo.toml`
   - `appVersion` in `charts/kueue/Chart.yaml`
   - `SECURITY-INSIGHTS.yaml` values by running `make update-security-insights GIT_TAG=$VERSION`
+  - Submit a pull request with the changes: <!-- example #3007 -->
+  - Cherry-pick the pull request onto the release branch
```
Comment (alculquicondor) [inline .github/ISSUE_TEMPLATE/NEW_RELEASE.md:64]: Did you mean `website` branch?

Comment (mimowo) [reply .github/ISSUE_TEMPLATE/NEW_RELEASE.md:64]: Yeah, I meant the analog of https://github.com/kubernetes-sigs/kueue/pull/2556

File: .github/ISSUE_TEMPLATE/NEW_RELEASE.md (hunk 2 above)
Comment (mimowo) [reply .github/ISSUE_TEMPLATE/NEW_RELEASE.md:56]: yeah, I did it all manually

Comment (mimowo) [reply .github/ISSUE_TEMPLATE/NEW_RELEASE.md]: Updated the steps, PTAL

File: .github/ISSUE_TEMPLATE/NEW_RELEASE.md [hunk 4]
```diff
@@ -24,23 +24,24 @@ Please do not remove items from the checklist
...
+      `git push upstream $VERSION`
   - Triggers prow to build and publish a staging container image
       `us-central1-docker.pkg.dev/k8s-staging-images/kueue/kueue:$VERSION`
+- [ ] An OWNER [prepares a draft release](https://github.com/kubernetes-sigs/kueue/releases)
+  - [ ] Create the draft release poiting out to the created tag.
+  - [ ] Write the change log into the draft release.
+  - [ ] Run
+      `make artifacts IMAGE_REGISTRY=registry.k8s.io/kueue GIT_TAG=$VERSION`
+      to generate the artifacts and upload the files in the `artifacts` folder
+      to the draft release.
```
Comment (alculquicondor) [inline .github/ISSUE_TEMPLATE/NEW_RELEASE.md:41]: Add `gh release upload <tag> artifacts/*` as an optional way of uploading the files.

File: .github/ISSUE_TEMPLATE/NEW_RELEASE.md [hunk 5]
```diff
@@ -49,13 +50,11 @@ Please do not remove items from the checklist
 - [ ] Run the [SBOM action](https://github.com/kubernetes-sigs/kueue/actions/workflows/sbom.yaml) to generate the SBOM and add it to the release.
 - [ ] For major or minor releases, merge the `main` branch into the `website` branch to publish the updated documentation.
 - [ ] Send an announcement email to `sig-scheduling@kubernetes.io` and `wg-batch@kubernetes.io` with the subject `[ANNOUNCE] kueue $VERSION is released`.   <!--Link: example https://groups.google.com/a/kubernetes.io/g/wg-batch/c/-gZOrSnwDV4 -->
-- [ ] Update the below files with respective values in `main` branch :
-  - Latest version in `README.md`
-  - Latest version in `cmd/experimental/kjobctl/docs/installation.md`
-  - Release notes in the `CHANGELOG`
-  - `version` in `site/hugo.toml`
-  - `appVersion` in `charts/kueue/Chart.yaml`
-  - `SECURITY-INSIGHTS.yaml` values by running `make update-security-insights GIT_TAG=$VERSION`
```
Comment (tenzen-y) [inline .github/ISSUE_TEMPLATE/NEW_RELEASE.md:58]: I guess that this is still needed, right?

File: .github/ISSUE_TEMPLATE/NEW_RELEASE.md [hunk 6]
```diff
@@ -49,13 +50,11 @@ Please do not remove items from the checklist
 - [ ] Run the [SBOM action](https://github.com/kubernetes-sigs/kueue/actions/workflows/sbom.yaml) to generate the SBOM and add it to the release.
 - [ ] For major or minor releases, merge the `main` branch into the `website` branch to publish the updated documentation.
 - [ ] Send an announcement email to `sig-scheduling@kubernetes.io` and `wg-batch@kubernetes.io` with the subject `[ANNOUNCE] kueue $VERSION is released`.   <!--Link: example https://groups.google.com/a/kubernetes.io/g/wg-batch/c/-gZOrSnwDV4 -->
```
Comment (alculquicondor) [inline .github/ISSUE_TEMPLATE/NEW_RELEASE.md:52]: This should probably be after the website is updated

File: .github/ISSUE_TEMPLATE/NEW_RELEASE.md [hunk 7]
```diff
@@ -49,13 +50,11 @@ Please do not remove items from the checklist
...
-  - Latest version in `cmd/experimental/kjobctl/docs/installation.md`
-  - Release notes in the `CHANGELOG`
-  - `version` in `site/hugo.toml`
-  - `appVersion` in `charts/kueue/Chart.yaml`
-  - `SECURITY-INSIGHTS.yaml` values by running `make update-security-insights GIT_TAG=$VERSION`
+- [ ] Update the `main` branch :
+  - [ ] Update `RELEASE_VERSION` in `Makefile` and run `make prepare-release-branch`
+  - [ ] Release notes in the `CHANGELOG`
+  - [ ] Submit a pull request with the changes: <!-- example #3007 -->
+  - [ ] Cherry-pick the pull request onto the website branch
```
Comment (alculquicondor) [inline .github/ISSUE_TEMPLATE/NEW_RELEASE.md:57]:
```suggestion
  - [ ] Cherry-pick the pull request onto the `website` branch
```

File: .github/ISSUE_TEMPLATE/NEW_RELEASE.md (hunk 4 above)
Comment (tenzen-y) [reply .github/ISSUE_TEMPLATE/NEW_RELEASE.md:41]: In the future, we may want to add a Make target to upload the artifacts using gh command.

File: .github/ISSUE_TEMPLATE/NEW_RELEASE.md (hunk 5 above)
Comment (alculquicondor) [reply .github/ISSUE_TEMPLATE/NEW_RELEASE.md:58]: oh yes, SECURITY_INSIGHTS is not fixed by `prepare-release-branch`. I suppose we could add a script to automate it, but maybe it's not worth yet.

File: .github/ISSUE_TEMPLATE/NEW_RELEASE.md (hunk 6 above)
Comment (mimowo) [reply .github/ISSUE_TEMPLATE/NEW_RELEASE.md:52]: +1

File: .github/ISSUE_TEMPLATE/NEW_RELEASE.md (hunk 5 above)
Comment (mimowo) [reply .github/ISSUE_TEMPLATE/NEW_RELEASE.md:58]: Right, added it back

Comment (tenzen-y) [reply .github/ISSUE_TEMPLATE/NEW_RELEASE.md:58]: I think so too.

File: .github/ISSUE_TEMPLATE/NEW_RELEASE.md (hunk 7 above)
Comment (mimowo) [reply .github/ISSUE_TEMPLATE/NEW_RELEASE.md]: Done

File: .github/ISSUE_TEMPLATE/NEW_RELEASE.md (hunk 6 above)
Comment (mimowo) [reply .github/ISSUE_TEMPLATE/NEW_RELEASE.md]: Moved it down.

File: .github/ISSUE_TEMPLATE/NEW_RELEASE.md (hunk 4 above)
Comment (mimowo) [reply .github/ISSUE_TEMPLATE/NEW_RELEASE.md]: Added the note about `gh release upload <tag> artifacts/*` in a separate point.

**Comment by alculquicondor** [comment]
Wonderful, thanks for the updates!

/lgtm

**Review by tenzen-y — COMMENTED** [review]
Thank you!
/lgtm

**Pull request merged**